cg/cli/upload.py

```diff
@@ -190,6 +190,29 @@ def _update_delivery_report_date(status_api, family_id):
     status_api.commit()
 
 
+@upload.command('delivery-report-to-scout')
+@click.argument('case_id')
+@click.option('-d', '--dry-run', 'dry_run', is_flag=True, help='run command without uploading to scout')
+@click.pass_context
+def delivery_report_to_scout(context, case_id, dry_run):
+    """Fetches an delivery-report from housekeeper and uploads it to scout"""
+    hk_api = context.obj['housekeeper_api']
+    report = _get_delivery_report_from_hk(hk_api, case_id)
+
+    LOG.info("uploading delivery report %s to scout for case: %s", report, case_id)
+    if not dry_run:
+        _add_delivery_report_to_scout(context, report, case_id)
+
+def _get_delivery_report_from_hk(hk_api: hk.HousekeeperAPI, family_id):
+    delivery_report_tag_name = 'delivery-report'
+    version_obj = hk_api.last_version(family_id)
+    uploaded_delivery_report_files = hk_api.get_files(bundle=family_id,
+                                                      tags=[delivery_report_tag_name],
+                                                      version=version_obj.id)
+
+    return uploaded_delivery_report_files[0].full_path
```

Comment:
Does this function raise an exception when the delivery report is not found or does it return `None`? If it is the latter, can we raise an exception?

Reply:
We will have an exception at this point if no delivery-report exists

Reply:
Will the return raise an exception because the list has no items, if so, I'd prefer an exception that says no delivery report is found instead.

Reply:
fixed